EIPS/eip-custom_errors_for_erc_tokens.md

```diff
@@ -0,0 +1,393 @@
+
+Indicates a failure with the token sender.
+Used in transfers.
+
+- MUST be used for disallowed transfers from the zero address.
+- MUST NOT be used for approval operations.
+- MUST NOT be used for balance or allowance requirements.
+  - Use `ERC1155InsufficientBalance` or `ERC1155InsufficientApproval` instead.
+
+#### `ERC1155InvalidReceiver(address receiver)`
+
+Indicates a failure with the token receiver.
+Used in transfers.
+
+- MUST be used for disallowed transfers to the zero address.
+- MUST be used for disallowed transfers to non-ERC1155TokenReceiver contracts or those that reject a transfer. (eg. returning an invalid response in `onERC1155Received`).
+- MUST NOT be used for approval operations.
+
+#### `ERC1155InsufficientApproval(address operator, uint256 id)`
+
+Indicates a failure with the `operator`'s approval in a transfer.
+Used in transfers.
+
+- MUST be used when operator `isApprovedForAll(owner, operator, id)` is false.
+
+#### `ERC1155InvalidApprover(address approver)`
+
+Indicates a failure with the `approver` of a token to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals from the zero address.
+- MUST NOT be used for transfer operations.
+
+#### `ERC1155InvalidOperator(address operator)`
+
+Indicates a failure with the `operator` to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals to the zero address.
+- MUST be used for disallowed approvals to the owner itself.
+- MUST NOT be used for transfer operations.
+  - Use `ERC1155InsufficientApproval` instead.
+
+#### `ERC1155InvalidArrayLength(uint256 idsLength, uint256 valuesLength)`
+
+Indicates an array length mismatch between `_ids` and `_values` in a `safeBatchTransferFrom` operation.
+Used in batch transfers.
+
+- MUST be used only if `idsLength` is different to `valuesLength`
+
+### Parameter Glossary
+
+| Name        | Description                                                                         |
+| ----------- | ----------------------------------------------------------------------------------- |
+| `sender`    | The address of whose token(s) (is/are) transferred from.                            |
+| `balance`   | Current balance for the interacting account.                                        |
+| `needed`    | Amount minimum required to perform an action.                                       |
+| `receiver`  | Address owner of the token(s) transferred.                                          |
+| `spender`   | Address not owner of the token(s) transferred, but allowed to operate on (it/them). |
+| `allowance` | Amount of token(s) a `spender` is allowed to operate with.                          |
+| `approver`  | Owner of the token(s) being approved to an `spender`.                               |
+| `tokenId`   | The identifier number of a token type.                                              |
+| `owner`     | Address of the owner of a token type.                                               |
```

Comment:
I think `token` vs `tokenType` is the same case as with `tokens` vs `token(s)`

The former is already considering all variants, either one token or many since it's plural and we kept `tokens` in other paragraphs.

I'd root for keeping `toke type`  or `token` instead of `token or token type` since they represent the same.
I'm particularly biased towards `token type`.

Reply:
Went for `token` only after sync meeting.